unlock-app/src/services/web3Service.js

```diff
@@ -516,8 +507,16 @@ export default class Web3Service extends EventEmitter {
       this.web3.eth.getTransaction(transactionHash),
     ]).then(([blockNumber, blockTransaction]) => {
       if (!blockTransaction) {
-        return this.emit('error', new Error(MISSING_TRANSACTION))
+        // The transaction is still pending: it has been sent to the network but not
+        // necessarily received by the node we're asking it (and not mined...)
+        this._watchTransaction(transactionHash)
+        return this.emit('transaction.updated', transactionHash, {
+          status: 'submitted',
+          confirmations: 0,
+          blockNumber: Number.MAX_SAFE_INTEGER,
```

Comment:
Would it be better to emit this without a blockNumber? Is there a downstream consumer with an expectation that this would be set?

Reply:
So, the goal of adding a very large number is because we sort things (like locks) based on their recency. If we put no blockNumber I guess that would be at the bottom... but maybe it is worth changing the sorting function?